catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/history/Historian.java

```diff
@@ -131,215 +112,110 @@ public UpdateResponse version(UpdateResponse updateResponse)
-                    new HashMap<>());
-            setSkipFlag(createStorageRequest);
-            CreateStorageResponse createStorageResponseResult = executeAsSystem(() -> {
-                CreateStorageResponse createStorageResponse = storageProvider().create(
-                        createStorageRequest);
-                storageProvider().commit(createStorageRequest);
-                overwriteResourceUris(createStorageResponse);
-                return createStorageResponse;
-            });
-
-            return Optional.ofNullable(createStorageResponseResult)
-                    .map(CreateStorageResponse::getProcessingErrors)
-                    .map(Set::isEmpty)
-                    .orElse(false);
-        });
-
-        preStaged.add(() -> {
-            Map<String, Serializable> props = new HashMap<>();
-            props.put(SKIP_VERSIONING, true);
-            CreateResponse createResponse =
-                    executeAsSystem(() -> catalogProvider().create(new CreateRequestImpl(
-                            versionedMetacards,
-                            props)));
-            return Optional.ofNullable(createResponse)
-                    .map(CreateResponse::getProcessingErrors)
-                    .map(Set::isEmpty)
-                    .orElse(false);
-        });
-
-        return Optional.of(transactionKey);
+        setResourceUriForContent(/*mutable*/ versionMetacards, createStorageResponse);
+
+        CreateResponse createResponse = storeVersionMetacards(versionMetacards);
+
+        return updateStorageResponse;
     }
 
     /**
      * Versions deleted {@link Metacard}s.
      *
      * @param deleteResponse Versions this responses deleted metacards
      */
-    public void version(DeleteResponse deleteResponse) {
+    public DeleteResponse version(DeleteResponse deleteResponse)
+            throws SourceUnavailableException, IngestException {
         if (doSkip(deleteResponse)) {
-            return;
+            return deleteResponse;
         }
         setSkipFlag(deleteResponse);
 
-        List<Metacard> versionedMetacards = deleteResponse.getDeletedMetacards()
+        Map<String, List<ContentItem>> contentItems = getContentItems(deleteResponse);
+        Action action = contentItems.isEmpty() ? Action.DELETED : Action.DELETED_CONTENT;
+
+        Map<String, Metacard> versionedMap =
+                getVersionMetacards(deleteResponse.getDeletedMetacards(),
+                        action,
+                        (Subject) deleteResponse.getRequest()
+                                .getProperties()
+                                .get(SecurityConstants.SECURITY_SUBJECT));
+
+        CreateStorageResponse createStorageResponse = versionContentItems(contentItems,
+                versionedMap);
+        if (createStorageResponse != null) {
+            setResourceUriForContent(/*Mutable*/ versionedMap, createStorageResponse);
+        }
+
+        CreateResponse createResponse =
```

Comment:
Not important, but you're never using this object; you only care about the creation of it.